From 806a016999ed53f375b48035ad58742157f7d556 Mon Sep 17 00:00:00 2001
From: "Axel Garcia K." <git@axelgk.com>
Date: Wed, 6 May 2026 08:29:07 +0000
Subject: [PATCH 2/4] Skip PicoDrive CHD media build

Add a NO_CD_MEDIA switch to the PicoDrive libretro Makefile and use it to omit the libchdr/CHD dependency stack for UniFrog builds.

The UniFrog PicoDrive module currently advertises cartridge-style extensions only, so keeping CHD support in the build adds cold-build work without producing a usable packaged format. MP3/Ogg handlers remain enabled because the core still references those symbols from the static CD code path.
---
 Makefile | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Makefile b/Makefile
index 601fd90e..82a74058 100644
--- a/Makefile
+++ b/Makefile
@@ -86,6 +86,7 @@ endif
 CFLAGS += -DREVISION=\"$(GIT_REVISION)\"
 
 # default settings
+NO_CD_MEDIA ?= 0
 use_libchdr ?= 1
 ifeq "$(ARCH)" "arm"
 use_cyclone ?= 1
@@ -345,6 +346,7 @@ else
 LDLIBS += -lvorbisfile -lvorbis -logg
 endif
 
+ifneq "$(NO_CD_MEDIA)" "1"
 ifeq (1,$(use_libchdr))
 CFLAGS += -DUSE_LIBCHDR
 
@@ -380,6 +382,7 @@ endif
 # ouf... prepend includes to overload headers available in the toolchain
 CFLAGS := -I$(LZMA)/include -I$(CHDR)/include $(CFLAGS)
 endif
+endif
 
 ifeq "$(PLATFORM_ZLIB)" "1"
 # zlib
